src/world/dimension_data.cc

```diff
@@ -477,39 +477,40 @@ namespace mcpe_viz {
         int16_t* emuchunk = new int16_t[NUM_BYTES_CHUNK_V3];
 
         // create png helpers
-        PngWriter png[(MAX_BLOCK_HEIGHT - MIN_BLOCK_HEIGHT) + 1];
-        for (int32_t cy = MIN_BLOCK_HEIGHT; cy <= MAX_BLOCK_HEIGHT; cy++) {
+        int dimensionBottomY = control.dimYBottom[dimId];
+        int dimensionTopY = control.dimYTop[dimId];
+        PngWriter png[(dimensionTopY - dimensionBottomY) + 1];
```

Comment:
For windows (VS), this is invalid as dynamic arrays can't be defined as such.
This seems to work (with basic testing):
```suggestion
        PngWriter *  png = new PngWriter[(dimensionTopY - dimensionBottomY) + 1];
```
```suggestion
        PngWriter png[(dimensionTopY - dimensionBottomY) + 1];
```